{"uid":"359cf47dbb9d7231","name":"Test Case 14 : News Latter Subscription Testing","fullName":"Tests.HomePage_ContactUs.Footer.test_Footer.TestFooter#test_news_latter_subscription","historyId":"bb69a50324e05f1e2c32236bff91ec08","time":{"start":1689848856496,"stop":1689848866975,"duration":10479},"descriptionHtml":"\n    <br><b>Test Case Description:</b>\n    <br><br>Validate the footer page is having subscribe filed  is taking the data and showing output like thank you message .\n    \n    <br><br><b>Expeceted Results:</b>\n    <br><br>Subcribe should be taking the data and gives with reply of thanks message.\n    \n    ","status":"broken","statusMessage":"TypeError: a bytes-like object is required, not 'method'","statusTrace":"self = <test_Footer.TestFooter object at 0x7f1f94e6c3a0>\n\n    @allure.title(\"Test Case 14 : News Latter Subscription Testing\")\n    @allure.description_html(\"\"\"\n    <br><b>Test Case Description:</b>\n    <br><br>Validate the footer page is having subscribe filed  is taking the data and showing output like thank you message .\n    \n    <br><br><b>Expeceted Results:</b>\n    <br><br>Subcribe should be taking the data and gives with reply of thanks message.\n    \n    \"\"\")\n    def test_news_latter_subscription(self):\n        try:\n>           news_latter_subscription_status = TestFooter.FooterObj.get_news_latter_subscription_status()\n\nTests/HomePage_ContactUs/Footer/test_Footer.py:57: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nPages/HomePage_ContactUs/Footer/Footer.py:59: in get_news_latter_subscription_status\n    subscription_done_msg = self.get_text_from_element(FooterElements.subscription_success)\nPages/BasePage.py:136: in get_text_from_element\n    element = self.wait.until(EC.visibility_of_element_located(by_locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ea69ef84e02dce2e66690247c3d03dde\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x7f1f94c46560>, message = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5626f9c24133 <unknown>\nE       #1 0x5626f9958966 <unknown>\nE       #2 0x5626f99950dc <unknown>\nE       #3 0x5626f9995211 <unknown>\nE       #4 0x5626f99cf6c4 <unknown>\nE       #5 0x5626f99b4bfd <unknown>\nE       #6 0x5626f99cd1c7 <unknown>\nE       #7 0x5626f99b49a3 <unknown>\nE       #8 0x5626f998946a <unknown>\nE       #9 0x5626f998a55e <unknown>\nE       #10 0x5626f9be3cae <unknown>\nE       #11 0x5626f9be78fe <unknown>\nE       #12 0x5626f9bf0f20 <unknown>\nE       #13 0x5626f9be8923 <unknown>\nE       #14 0x5626f9bbbc0e <unknown>\nE       #15 0x5626f9c0bb08 <unknown>\nE       #16 0x5626f9c0bc97 <unknown>\nE       #17 0x5626f9c1c113 <unknown>\nE       #18 0x7f25e6c94b43 <unknown>\n\n/usr/local/lib/python3.10/dist-packages/selenium/webdriver/support/wait.py:95: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_Footer.TestFooter object at 0x7f1f94e6c3a0>\n\n    @allure.title(\"Test Case 14 : News Latter Subscription Testing\")\n    @allure.description_html(\"\"\"\n    <br><b>Test Case Description:</b>\n    <br><br>Validate the footer page is having subscribe filed  is taking the data and showing output like thank you message .\n    \n    <br><br><b>Expeceted Results:</b>\n    <br><br>Subcribe should be taking the data and gives with reply of thanks message.\n    \n    \"\"\")\n    def test_news_latter_subscription(self):\n        try:\n            news_latter_subscription_status = TestFooter.FooterObj.get_news_latter_subscription_status()\n    \n            assert news_latter_subscription_status\n        except:\n>           allure.attach(self.FooterObj.driver.get_screenshot_as_png,name=\"Footer Element Error\", attachment_type= allure.attachment_type.PNG)\n\nTests/HomePage_ContactUs/Footer/test_Footer.py:61: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/lib/python3.10/site-packages/allure_commons/_allure.py:200: in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n../../../.local/lib/python3.10/site-packages/pluggy/_hooks.py:265: in __call__\n    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)\n../../../.local/lib/python3.10/site-packages/pluggy/_manager.py:80: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n../../../.local/lib/python3.10/site-packages/allure_pytest/listener.py:229: in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n../../../.local/lib/python3.10/site-packages/allure_commons/reporter.py:165: in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n../../../.local/lib/python3.10/site-packages/pluggy/_hooks.py:265: in __call__\n    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)\n../../../.local/lib/python3.10/site-packages/pluggy/_manager.py:80: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <allure_commons.logger.AllureFileLogger object at 0x7f1fbb182230>\nbody = <bound method WebDriver.get_screenshot_as_png of <undetected_chromedriver.Chrome (session=\"ea69ef84e02dce2e66690247c3d03dde\")>>\nfile_name = '7b9bd9d3-bf65-4dd7-b919-c812f02085fc-attachment.png'\n\n    @hookimpl\n    def report_attached_data(self, body, file_name):\n        destination = os.path.join(self._report_dir, file_name)\n        with open(destination, 'wb') as attached_file:\n            if isinstance(body, text_type):\n                attached_file.write(body.encode('utf-8'))\n            else:\n>               attached_file.write(body)\nE               TypeError: a bytes-like object is required, not 'method'\n\n../../../.local/lib/python3.10/site-packages/allure_commons/logger.py:62: TypeError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"init_driver","time":{"start":1689848836256,"stop":1689848837761,"duration":1505},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":false}],"testStage":{"descriptionHtml":"\n    <br><b>Test Case Description:</b>\n    <br><br>Validate the footer page is having subscribe filed  is taking the data and showing output like thank you message .\n    \n    <br><br><b>Expeceted Results:</b>\n    <br><br>Subcribe should be taking the data and gives with reply of thanks message.\n    \n    ","status":"broken","statusMessage":"TypeError: a bytes-like object is required, not 'method'","statusTrace":"self = <test_Footer.TestFooter object at 0x7f1f94e6c3a0>\n\n    @allure.title(\"Test Case 14 : News Latter Subscription Testing\")\n    @allure.description_html(\"\"\"\n    <br><b>Test Case Description:</b>\n    <br><br>Validate the footer page is having subscribe filed  is taking the data and showing output like thank you message .\n    \n    <br><br><b>Expeceted Results:</b>\n    <br><br>Subcribe should be taking the data and gives with reply of thanks message.\n    \n    \"\"\")\n    def test_news_latter_subscription(self):\n        try:\n>           news_latter_subscription_status = TestFooter.FooterObj.get_news_latter_subscription_status()\n\nTests/HomePage_ContactUs/Footer/test_Footer.py:57: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \nPages/HomePage_ContactUs/Footer/Footer.py:59: in get_news_latter_subscription_status\n    subscription_done_msg = self.get_text_from_element(FooterElements.subscription_success)\nPages/BasePage.py:136: in get_text_from_element\n    element = self.wait.until(EC.visibility_of_element_located(by_locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ea69ef84e02dce2e66690247c3d03dde\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x7f1f94c46560>, message = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x5626f9c24133 <unknown>\nE       #1 0x5626f9958966 <unknown>\nE       #2 0x5626f99950dc <unknown>\nE       #3 0x5626f9995211 <unknown>\nE       #4 0x5626f99cf6c4 <unknown>\nE       #5 0x5626f99b4bfd <unknown>\nE       #6 0x5626f99cd1c7 <unknown>\nE       #7 0x5626f99b49a3 <unknown>\nE       #8 0x5626f998946a <unknown>\nE       #9 0x5626f998a55e <unknown>\nE       #10 0x5626f9be3cae <unknown>\nE       #11 0x5626f9be78fe <unknown>\nE       #12 0x5626f9bf0f20 <unknown>\nE       #13 0x5626f9be8923 <unknown>\nE       #14 0x5626f9bbbc0e <unknown>\nE       #15 0x5626f9c0bb08 <unknown>\nE       #16 0x5626f9c0bc97 <unknown>\nE       #17 0x5626f9c1c113 <unknown>\nE       #18 0x7f25e6c94b43 <unknown>\n\n/usr/local/lib/python3.10/dist-packages/selenium/webdriver/support/wait.py:95: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_Footer.TestFooter object at 0x7f1f94e6c3a0>\n\n    @allure.title(\"Test Case 14 : News Latter Subscription Testing\")\n    @allure.description_html(\"\"\"\n    <br><b>Test Case Description:</b>\n    <br><br>Validate the footer page is having subscribe filed  is taking the data and showing output like thank you message .\n    \n    <br><br><b>Expeceted Results:</b>\n    <br><br>Subcribe should be taking the data and gives with reply of thanks message.\n    \n    \"\"\")\n    def test_news_latter_subscription(self):\n        try:\n            news_latter_subscription_status = TestFooter.FooterObj.get_news_latter_subscription_status()\n    \n            assert news_latter_subscription_status\n        except:\n>           allure.attach(self.FooterObj.driver.get_screenshot_as_png,name=\"Footer Element Error\", attachment_type= allure.attachment_type.PNG)\n\nTests/HomePage_ContactUs/Footer/test_Footer.py:61: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/lib/python3.10/site-packages/allure_commons/_allure.py:200: in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n../../../.local/lib/python3.10/site-packages/pluggy/_hooks.py:265: in __call__\n    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)\n../../../.local/lib/python3.10/site-packages/pluggy/_manager.py:80: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n../../../.local/lib/python3.10/site-packages/allure_pytest/listener.py:229: in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n../../../.local/lib/python3.10/site-packages/allure_commons/reporter.py:165: in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n../../../.local/lib/python3.10/site-packages/pluggy/_hooks.py:265: in __call__\n    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)\n../../../.local/lib/python3.10/site-packages/pluggy/_manager.py:80: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <allure_commons.logger.AllureFileLogger object at 0x7f1fbb182230>\nbody = <bound method WebDriver.get_screenshot_as_png of <undetected_chromedriver.Chrome (session=\"ea69ef84e02dce2e66690247c3d03dde\")>>\nfile_name = '7b9bd9d3-bf65-4dd7-b919-c812f02085fc-attachment.png'\n\n    @hookimpl\n    def report_attached_data(self, body, file_name):\n        destination = os.path.join(self._report_dir, file_name)\n        with open(destination, 'wb') as attached_file:\n            if isinstance(body, text_type):\n                attached_file.write(body.encode('utf-8'))\n            else:\n>               attached_file.write(body)\nE               TypeError: a bytes-like object is required, not 'method'\n\n../../../.local/lib/python3.10/site-packages/allure_commons/logger.py:62: TypeError","steps":[{"name":"Taking News Latter Subscription Elements","time":{"start":1689848856496,"stop":1689848866974,"duration":10478},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5626f9c24133 <unknown>\n#1 0x5626f9958966 <unknown>\n#2 0x5626f99950dc <unknown>\n#3 0x5626f9995211 <unknown>\n#4 0x5626f99cf6c4 <unknown>\n#5 0x5626f99b4bfd <unknown>\n#6 0x5626f99cd1c7 <unknown>\n#7 0x5626f99b49a3 <unknown>\n#8 0x5626f998946a <unknown>\n#9 0x5626f998a55e <unknown>\n#10 0x5626f9be3cae <unknown>\n#11 0x5626f9be78fe <unknown>\n#12 0x5626f9bf0f20 <unknown>\n#13 0x5626f9be8923 <unknown>\n#14 0x5626f9bbbc0e <unknown>\n#15 0x5626f9c0bb08 <unknown>\n#16 0x5626f9c0bc97 <unknown>\n#17 0x5626f9c1c113 <unknown>\n#18 0x7f25e6c94b43 <unknown>\n\n","statusTrace":"  File \"/home/kishorekumarmahto/.local/lib/python3.10/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"/home/kishorekumarmahto/my_backup/Testing/Dr._Garcia_Biomagnetism_website/Pages/HomePage_ContactUs/Footer/Footer.py\", line 59, in get_news_latter_subscription_status\n    subscription_done_msg = self.get_text_from_element(FooterElements.subscription_success)\n  File \"/home/kishorekumarmahto/.local/lib/python3.10/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"/home/kishorekumarmahto/my_backup/Testing/Dr._Garcia_Biomagnetism_website/Pages/BasePage.py\", line 136, in get_text_from_element\n    element = self.wait.until(EC.visibility_of_element_located(by_locator))\n  File \"/usr/local/lib/python3.10/dist-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[{"name":"Entering Email Id For News Latter Subscription","time":{"start":1689848856496,"stop":1689848856563,"duration":67},"status":"passed","steps":[{"name":"Sending Text To The Input Box","time":{"start":1689848856496,"stop":1689848856563,"duration":67},"status":"passed","steps":[],"attachments":[],"parameters":[{"name":"by_locator","value":"('xpath', \"(//input[@id='wpforms-10-field_2'])[1]\")"},{"name":"text","value":"'lokkemofya@gufum.com'"}],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":true}],"attachments":[],"parameters":[],"stepsCount":1,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":true},{"name":"Clicked On Subscribe Button","time":{"start":1689848856563,"stop":1689848856611,"duration":48},"status":"passed","steps":[{"name":"Clicked On Element","time":{"start":1689848856563,"stop":1689848856611,"duration":48},"status":"passed","steps":[],"attachments":[],"parameters":[{"name":"by_locator","value":"('xpath', \"(//button[normalize-space()='Subscribe'])[1]\")"}],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":true}],"attachments":[],"parameters":[],"stepsCount":1,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":true},{"name":"Taking Text from element","time":{"start":1689848856611,"stop":1689848866974,"duration":10363},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x5626f9c24133 <unknown>\n#1 0x5626f9958966 <unknown>\n#2 0x5626f99950dc <unknown>\n#3 0x5626f9995211 <unknown>\n#4 0x5626f99cf6c4 <unknown>\n#5 0x5626f99b4bfd <unknown>\n#6 0x5626f99cd1c7 <unknown>\n#7 0x5626f99b49a3 <unknown>\n#8 0x5626f998946a <unknown>\n#9 0x5626f998a55e <unknown>\n#10 0x5626f9be3cae <unknown>\n#11 0x5626f9be78fe <unknown>\n#12 0x5626f9bf0f20 <unknown>\n#13 0x5626f9be8923 <unknown>\n#14 0x5626f9bbbc0e <unknown>\n#15 0x5626f9c0bb08 <unknown>\n#16 0x5626f9c0bc97 <unknown>\n#17 0x5626f9c1c113 <unknown>\n#18 0x7f25e6c94b43 <unknown>\n\n","statusTrace":"  File \"/home/kishorekumarmahto/.local/lib/python3.10/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"/home/kishorekumarmahto/my_backup/Testing/Dr._Garcia_Biomagnetism_website/Pages/BasePage.py\", line 136, in get_text_from_element\n    element = self.wait.until(EC.visibility_of_element_located(by_locator))\n  File \"/usr/local/lib/python3.10/dist-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[{"name":"by_locator","value":"('xpath', '(//p[contains(text(),\"Thanks for signing up for the newsletter! We\\'ll be\")])[1]')"}],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":true,"hasContent":true}],"attachments":[],"parameters":[],"stepsCount":5,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":true}],"attachments":[{"uid":"628bc7fb1c44d434","name":"Footer Element Error","source":"628bc7fb1c44d434.png","type":"image/png","size":0}],"parameters":[],"stepsCount":6,"attachmentsCount":1,"shouldDisplayMessage":true,"hasContent":true},"afterStages":[{"name":"init_driver::0","time":{"start":1689848874803,"stop":1689848874858,"duration":55},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":false}],"labels":[{"name":"tag","value":"@pytest.mark.usefixtures('init_driver')"},{"name":"parentSuite","value":"Tests.HomePage_ContactUs.Footer"},{"name":"suite","value":"test_Footer"},{"name":"subSuite","value":"TestFooter"},{"name":"host","value":"isadmin"},{"name":"thread","value":"45264-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"Tests.HomePage_ContactUs.Footer.test_Footer"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"tags":["@pytest.mark.usefixtures('init_driver')"]},"source":"359cf47dbb9d7231.json","parameterValues":[]}